### PR TITLE
avro-cpp: 1.8.2 -> 1.10.0

### PR DESCRIPTION
**Pull request by smunix**
###### Motivation for this change

Upgrade to the latest version of avro-cpp, 1.10.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7093]: Please put this into a separate commit with the message ``maintainers: add smunix``.

File: pkgs/development/libraries/avro-c++/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/avro-c++/default.nix:1]: Can python2 be replaced?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/libraries/avro-c++/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/avro-c++/default.nix:32]:
```suggestion
    maintainers = with lib.maintainers; [ rasendubi smunix ];
    platforms = lib.platforms.all;
```

File: pkgs/development/libraries/avro-c++/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/avro-c++/default.nix:30]:
```suggestion
    license = lib.licenses.asl20;
```

Comment (rasendubi) [reply pkgs/development/libraries/avro-c++/default.nix:30]: This won't eval as `lib` is not in scope.

**Comment by rasendubi** [comment]
I know this has been waiting for a review way too long… sorry about that.

@smunix, do you still want to be co-maintainer of avro-cpp? I think the only critical comment is to separate `maintainers-list.nix` change, maybe bump version to 1.10.1 while you're at it.

If you don't have the motivation, I understand can do the package update for you.

**Comment by smunix** [comment]
> I know this has been waiting for a review way too long… sorry about that.
> 
> @smunix, do you still want to be co-maintainer of avro-cpp? I think the only critical comment is to separate `maintainers-list.nix` change, maybe bump version to 1.10.1 while you're at it.
> 
Sure, let's hop to `1.10.1`

> If you don't have the motivation, I understand and can do the package update for you.

Was busy with other projects too, I'll fix the package update this week.

File: pkgs/development/libraries/avro-c++/default.nix
Comment (SuperSandro2000) [reply pkgs/development/libraries/avro-c++/default.nix:30]: Then please add lib to inputs. This is a recent contributing guide change and stdenv.lib is deprecated.

**Pull request closed without merging**
